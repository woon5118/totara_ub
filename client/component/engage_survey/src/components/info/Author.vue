<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Kian Nguyen <kian.nguyen@totaralearning.com>
  @module engage_survey
-->

<template>
  <div class="tui-engageSurveyAuthor">
    <Avatar :src="profileImageUrl" :alt="profileImageAlt" size="xxsmall" />

    <a
      :href="$url('/user/profile.php', { id: userId })"
      class="tui-engageSurveyAuthor__userLink"
    >
      {{ fullname }}
    </a>
  </div>
</template>

<script>
import Avatar from 'tui/components/avatar/Avatar';

export default {
  components: {
    Avatar,
  },

  props: {
    userId: {
      required: true,
      type: [Number, String],
    },

    fullname: {
      required: true,
      type: String,
    },

    profileImageUrl: {
      required: true,
      type: String,
    },

    profileImageAlt: {
      required: true,
      type: String,
    },
  },
};
</script>

<style lang="scss">
.tui-engageSurveyAuthor {
  display: flex;
  align-items: center;

  &__userLink {
    margin-left: var(--gap-2);
    font-size: var(--font-size-13);
  }
}
</style>
