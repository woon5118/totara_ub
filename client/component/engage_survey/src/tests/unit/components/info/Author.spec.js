/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Kian Nguyen <kian.nguyen@totaralearning.com>
 * @module engage_article
 */

import Author from 'engage_survey/components/info/Author';
import { mount } from '@vue/test-utils';

describe('engage_survey/components/info/Author.vue', function() {
  let wrapper = null;

  beforeAll(function() {
    wrapper = mount(Author, {
      mocks: {
        $url(uri, params) {
          return `${uri}?${params.toString()}`;
        },
      },
      propsData: {
        userId: 42,
        fullname: 'Full name',
        profileImageUrl: 'http://example.com',
        profileImageAlt: 'Hello',
      },
    });
  });

  it('Checks snapshot', function() {
    expect(wrapper.element).toMatchSnapshot();
  });
});
