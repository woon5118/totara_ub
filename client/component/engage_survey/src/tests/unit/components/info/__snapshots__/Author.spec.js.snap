// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`engage_survey/components/info/Author.vue Checks snapshot 1`] = `
<div
  class="tui-engageSurveyAuthor"
>
  <img
    alt="Hello"
    class="tui-avatar tui-avatar--xxsmall"
    src="http://example.com"
  />
   
  <a
    class="tui-engageSurveyAuthor__userLink"
    href="/user/profile.php?[object Object]"
  >
    
    Full name
  
  </a>
</div>
`;
